.github/workflows/test-action.yml

```diff
@@ -0,0 +1,118 @@
+name: Test GitHub Action
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
+
+jobs:
+  test-defaults:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./output/asyncapi.md" ]; then
+            echo "Files exist"
+          else
+            echo "Files do not exist:- ./output/asyncapi.md"
+            echo "Action failed"
+            exit 1
+          fi
+
+  test-validate-success:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+          command: validate
+
+  test-custom-command:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          custom_command: "generate models typescript ./test/asyncapi.yml -o ./output"
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./output/AnonymousSchema_1.ts" ]; then
+            echo "Models have been generated"
+          else
+            echo "Models have not been generated"
+            echo "Action failed"
+            exit 1
+          fi
+  
+  test-custom-output:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+          output: custom-output
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./custom-output/asyncapi.md" ]; then
+            echo "Files exist"
+          else
+            echo "Files do not exist:- ./custom-output/asyncapi.md"
+            echo "Action failed"
+            exit 1
+          fi
+  
+  test-file-not-found:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        id: test
+        uses: ./
+        with:
+          filepath: non_existent_file.yml
+        continue-on-error: true
+      - name: Check for failure
+        run: if [ "${{ steps.test.outcome }}" == "success" ]; then exit 1; fi
+  
+  test-invalid-input:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        id: test
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+          command: generate # No template or language specified
+          template: '' # Empty string
+        continue-on-error: true
+      - name: Check for failure
+        run: if [ "${{ steps.test.outcome }}" == "success" ]; then exit 1; fi
+
+  send-to-slack:
```

Comment:
this workflow should run only on PR, so no need to drop errors to slack as they will be visible on PR

like https://github.com/asyncapi/github-action-for-generator/actions/runs/7304150811/job/20169330511?pr=348